staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go

```diff
@@ -512,3 +516,38 @@ func TestHistogramVec_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestGetHistogramFromGatherer(t *testing.T) {
+	// Have multiple labels defined here, but we only care about the first label, i.e. "extension_point".
+	labels := []string{"extension_point", "status"}
+	lvMap := map[string]string{"extension_point": "*"}
+	HistogramOpts := &metrics.HistogramOpts{
+		Namespace: "namespace",
+		Name:      "metric_test_name",
+		Subsystem: "subsystem",
+		Help:      "histogram help message",
+		Buckets:   prometheus.DefBuckets,
+	}
+	vec := metrics.NewHistogramVec(HistogramOpts, labels)
+	legacyregistry.MustRegister(vec)
+	vec.WithLabelValues("filter", "0").Observe(1.5)
+	vec.WithLabelValues("score", "0").Observe(3.5)
```

Comment:
Also add

```golang
	vec.WithLabelValues("filter", "1").Observe(1.5)
	vec.WithLabelValues("score", "1").Observe(3.5)
```

Reply:
hmm... I didn't aware of this, so this is the case this PR cannot address, I am incline to close this.

Reply:
You mean close this comment?

The scheduler produces observations like this. An extension_point=Filter can have a result Unschedulable or Scheduled or Error. We need the information averaged for all of them.

Reply:
> You mean close this comment?

no, I just think it will be a little complicate to achieve that with the pr like this one, I was thinking close this pr with another one which just fetch the filter and score,  should be easy to just call the existing `GetHistogramVecFromGatherer` twice, pls let me think about it.

/hold

Reply:
Fine with me